hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/BaseTableServicePlanActionExecutor.java

```diff
@@ -124,7 +124,9 @@ public Pair<Option<HoodieInstant>, Set<String>> getIncrementalPartitions(TableSe
         .filter(this::filterCommitByTableType).flatMap(instant -> {
           try {
             String completionTime = instant.getCompletionTime();
-            if (completionTime.compareTo(leftBoundary) >= 0 && completionTime.compareTo(rightBoundary) < 0) {
+            if (completionTime.compareTo(leftBoundary) >= 0
+                && ((instant.requestedTime().length() < completionTime.length() && instant.requestedTime().compareTo(rightBoundary) < 0)
```

Comment:
This line is confusing, can we at least add a comment, or maybe abstract it as a separate method.

Reply:
yes, and lets add java docs.

Reply:
Not part of this PR. Removed it.